src/Steeltoe.Discovery.ClientCore/DiscoveryServiceCollectionExtensions.cs

```diff
@@ -142,6 +143,11 @@ private static void AddEurekaServices(IServiceCollection services, IDiscoveryLif
                 services.AddSingleton(lifecycle);
             }
 
+            if (handler != null)
+            {
+                services.AddTransient(s => new EurekaDiscoveryClientHandlerProvider(handler));
```

Comment:
I'm thinking we want to change this to inject a IDiscoveryClientHandlerProvider instead of  a EurekaDiscoveryClientHandlerProvider.

Also, this should probably be AddSingleton<IDiscoveryClientHandlerProvider>() instead of AddTransient().

Thoughts?

Reply:
I have changed implementation. HttpClient can dispose HttpHandler. I changed parameter to Func<HttpClientHandler> from HttpClientHandler.
EurekaDiscoveryClientHandlerProvider is registered as singleton.